Simplify kafka shutdown and default to IN_MEMORY

naddeoa opened the pull request.

Commit:
Simplify kafka consumer shutdown behavior and fix potential bug

The way I had it before, while cooler than it was, technically wasn't
correct. By shutting down the consumer in the poll loop I risked the
`close` being called on a separate thread, since each iteration called
`yeild` and potentially swapped threads to another in Dispatchers.IO.
Now, each consumer has an actual dedicated thread that it runs on with a
dedicated context and closing happens in the shutdown hook's Thread by
switching over to that context in `runBlocking` and executing the close
there.

The KotlinConsumer doesn't need to be a CoroutineScope at all anymore
and the worker launches don't need to be launched themselves. The
consumer is totally self contained in its scope with its shutdown logic
and the shutdown hook waits for the close.

Commit:
Allow specifying upload cadence and default to IN_MEMORY

The sqlite implementation of the profile store has this bad perf issue
that happens when the container runs long enough so I'm going to default
the store to in-memory instead. To compensate for the potential to lose
data via javalin crashes (already a very rare event), I added an option
to separately specify the cadence to upload profiles.

Now, you can have an hourly model whose profiles you upload every 5, 10,
30 minutes, for example. That mitigates the max amount of data you
might lose for a transient crash and doesn't take the massive perf hit
that you currently take defaulting to sqlite.

murilommen commented on src/main/kotlin/ai/whylabs/services/whylogs/core/config/EnvVars.kt:
```diff
@@ -18,9 +19,25 @@ enum class WriterTypes {
+            FIVE_MINUTES -> return Duration.of(5, ChronoUnit.MINUTES)
+            TEN_MINUTES -> return Duration.of(10, ChronoUnit.MINUTES)
+            THIRTY_MINUTES -> return Duration.of(30, ChronoUnit.MINUTES)
+            ON_DEMAND -> return Duration.of(1000, ChronoUnit.DAYS)
```
why ON_DEMAND needs to take an "infinite" amount of Duration in this case?

naddeoa replied:
Just updated the logic to be more clear. There isn't an INF duration unfortunately so I made it nullable.

murilommen reviewed (dismissed):
looks good. didn't catch anything weird, but left a comment out of curiosity :)

Commit:
Update ON_DEMAND logic to be more clear

Before ON_DEMAND had a really long duration and an if check to just
ignore it in the manager. Now it just has a null duration and the
manager ignores anything with a null duration.

Commit:
Use delimited writing for protobuf files to whylabs

This works around an issue that we have reading non-delimited protobuf
files. We'll also fix that in our backend but there's no harm in doing
this as well.
